financial-templates-lib/clients/Web3WebsocketClient.js

```diff
@@ -0,0 +1,34 @@
+// This client exports a web3 instances which mimics that created by the default truffle config. It exclusively uses
+// websocket connections to take advantage of reconnect logic if the websocket disconnects or times out. The network
+// syntax mimics that of the main UMA Truffle implementation to make this backwards compatible.
+
+const Web3 = require("web3");
+const { getTruffleConfig, nodeUrl } = require("@umaprotocol/common/TruffleConfig");
+const argv = require("minimist")(process.argv.slice(), { string: ["network"] });
+
+const websocketOptions = {
+  timeout: 10000, // ms
+  clientConfig: {
+    maxReceivedFrameSize: 100000000, // Useful if requests result are large bytes - default: 1MiB
+    maxReceivedMessageSize: 100000000 // bytes - default: 8MiB
+  },
+  reconnect: {
+    auto: true, // Enable auto reconnection
+    delay: 5000, // ms
+    maxAttempts: 10,
+    onTimeout: false
+  }
+};
+
+// Create websocket web3 provider. This contains the re-try logic on failed/timeout connections.
+const websocketProvider = new Web3.providers.WebsocketProvider(nodeUrl, websocketOptions);
+
+// Use the websocketProvider to create a provider with an unlocked wallet. This piggybacks off the UMA common TruffleConfig
+// implementing all networks & wallet types. EG mainnet_mnemonic, kovan_gckms, test, mainnet-fork. Errors if no argv.network.
+const walletWithWebsocketProvider = new getTruffleConfig().networks[argv.network].provider(websocketProvider);
```

Comment:
nit: does this need a `new`?

Reply:
I had the `new` here as the underlying syntax required a `new` but I think you are right that it's redundant as the `singletonProvider` from the `provider` function returns a `new` `singletonProvider`. 

After removing this seems like still works just fine.